qt/draw_widget.cpp

```diff
@@ -195,6 +214,35 @@ void DrawWidget::initializeGL()
     m_screenshoter->Start();
 }
 
+bool DrawWidget::event(QEvent * event)
+{
+  if (!isQtTouchEvent(event))
+    return QOpenGLWidget::event(event);
```

Comment:
Simpler is better here:
- remove isQtTouchEvent
- add df::TouchEvent::NONE and return it in qtTouchEventTypeToDfTouchEventType
- call qtTouchEventTypeToDfTouchEventType here once and check for df::TouchEvent::NONE

Reply:
Done